### ISSUE TITLE
Pick a custom timer time

### ISSUE THREAD

**Issue by mr-landerson1** [adrcotfas/goodtime#168]
Instead of having to go through the menu to change the time from 25 minutes to something else, when starting a new session, have the ability to modify the time for the pomodorro session.

For example, I am about to go to lunch and have 10 minutes to work on something. I should be able to select 10 minutes easily and start the timer.

**Comment by fusillijerry89** [comment]
Yes, I would like the ability to set a different amount of time for different labels. If I could make a new label, and have the length to be 10 minutes instead of the default 25 minutes, then I could use this for different types of habits.

**Comment by adrcotfas** [comment]
@mr-landerson1, @fusillijerry89 You are describing two different use cases.
1 is to easily change the duration from the main screen
2 is to have labels attached to duration profiles

For 1, one idea would be to implement a gesture like: long click on the timer to enter an edit mode, then swipe up or down to configure the timer. I'll consider this in future versions.

2 would imply not so straightforward changes (UI, functionality, database) and I don't think that it will come soon.

A quick alternative for both your proposals would be to create your favorite duration profiles and change them from the bottom right dialog (the one to switch the current label). That would take ~5 seconds.

**Comment by studgeek** [comment]
For 1 what I personally would like is ability to remove a minute with swipe. For example, one possibility would be that swipe down could remove a minute to be consistent with existing swipe up.  Stop session could move to double tap. I guess this could be done with an option to maintain backward compatibility.

**Comment by studgeek** [comment]
Here is an example of my last suggestion - https://github.com/studgeek/Goodtime/commit/286c6950653d17a22fb2d738af72f2843a7f036a

**Comment by tippfehlr** [comment]
https://github.com/adrcotfas/Goodtime/issues/217#issuecomment-713077735 sounds really nice ... I imagine an egg clock-like ux (I set a timer for how long I want to work, then that timer ticks down to 0)

I hope this is still considered after four years ^^

**Comment by adrcotfas** [comment]
The next major update will make it easier to start a custom countdown.
